operators/swap_strips.py

```diff
@@ -1,33 +1,62 @@
 import bpy
 from operator import attrgetter
+from .utils.find_closest_strip import find_closest_strip
 
 class SwapStrips(bpy.types.Operator):
     """
-    Swaps the 2 selected strips between them. More specific, places the first
-    strip in the channel and starting frame (frame_final_start) of the second
-    strip, and places the second strip in the channel and starting frame
-    (frame_final_end) of the first strip. If there is no space for the swap, it 
-    does nothing. If at least 1 of the selected strips is of effect type, the
-    operator won't have any effect.
+    Swaps the 2 strips between them. More specific, places the first strip in 
+    the channel and starting frame (frame_final_start) of the second strip, and 
+    places the second strip in the channel and starting frame (frame_final_end) 
+    of the first strip. If there is no space for the swap, it does nothing.
     """
     bl_idname = "power_sequencer.swap_strips"
     bl_label = "Swap Strips"
-    bl_description = "Swaps the 2 selected strips between them"
+    bl_description = "Swaps 2 strips between them"
     bl_options = {"REGISTER", "UNDO"}
+    
+    direction = bpy.props.EnumProperty(
+		name="Direction",
+		description="The direction to find the closest strip",
+		items=[("up", "Up", "The direction up from the selected strip"), \
+                ("down", "Down", "The direction down from the selected strip"), \
+                ("left", "Left", "The direction left from the selected strip"), \
+                ("right", "Right", "The direction right from the selected strip")],
+        default="up"
+	)
 
     @classmethod
     def poll(cls, context):
-        return len(bpy.context.selected_sequences) == 2
+        return len(bpy.context.selected_sequences) in range(1, 3)
 
     def execute(self, context):
         strip_1 = context.selected_sequences[0]
-        strip_2 = context.selected_sequences[1]
-        
+        if len(context.selected_sequences) == 1:
+            strip_2 = self.closest_direction_strip(context, \
+                    context.selected_sequences[0])
+            if None == strip_2:
+                return {'CANCELLED'}
+        else:

```

Comment:
No need for the else, you can dedent the following line

Reply:
The else here goes to the above if statement, and not: if None == strip_2:
I can't see how this else could be erased. Probably I'm missing something here..